apiserver/facades/client/spaces/spaces_internal_test.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2024 Canonical Ltd.
```

Comment:
We are missing a lot of testing here.

The uses of CreateSpace that in turn uses checkSupportsSpaces should be getting tested here not so much `checkSupportsSpace` as that is implementation specific.

We don't have any tests for esnureSpacesAreMutable as well. Are we able to move these over to mock tests?

Reply:
I see what is happening now.

Reply:
I just ported some of the tests over from the original test suite as it was too hard to change them to the new implementation with the state-based tests. Eventually all the tests should be moved over here, but I don't wanna do this all now.

Between this and the original test suite, do you see any missing coverage?